src/ansi-c/library/unistd.c

```diff
@@ -319,3 +319,27 @@ ret_type _read(int fildes, void *buf, size_type nbyte)
   __CPROVER_HIDE:;
   return read(fildes, buf, nbyte);
 }
+
+/* FUNCTION: sysconf */
+
+#ifndef __CPROVER_ERRNO_H_INCLUDED
+#  include <errno.h>
+#  define __CPROVER_ERRNO_H_INCLUDED
+#endif
+
+long __VERIFIER_nondet_long(void);
+__CPROVER_bool __VERIFIER_nondet___CPROVER_bool(void);
+
+long sysconf(int name);
+
+// This overapproximation is based on this specification
+// https://man7.org/linux/man-pages/man3/sysconf.3.html.
+long sysconf(int name)
+{
+__CPROVER_HIDE:;
+  (void)name;
+  long retval = __VERIFIER_nondet_long();
+  if(retval == -1 && __VERIFIER_nondet___CPROVER_bool())
```

Comment:
The doc for `errno` says the following: 
```
The value in errno is significant only when the return value of
       the call indicated an error (i.e., -1 from most system calls; -1
       or NULL from most library functions); a function that succeeds is
       allowed to change errno.  The value of errno is never set to zero
       by any system call or library function.
```
So it seems like this model should allow `errno` to change nondeterministically. Also in case of error is `EINVAL` really  the only possible error value ? If the client code has a branch that checks for another `errno` value this branch would be unreachable under this model.

Reply:
Can you get more info about possible `errno` values ?

Reply:
I added more comments to explain the rationale behind the implementation. According to the specification, `errno` will either be set to `EINVAL` or it will remain unchanged, even if the function return error (i.e., -1).

Reply:
The pubs.opengroup.org link gives two error cases:
- "invalid option name", set errno = EINVAL and return -1,
- "name corresponds to a limit and the limit is undefined", keep errno the same and return -1
And EINVAL is given as the only possible errno value, so this is OK.

But it also says:
> If the variable corresponding to name is dependent on an unsupported option, the results are unspecified.

And then in the example usage section:
> Application developers should check whether an option, such as _POSIX_TRACE, is supported prior to obtaining and using values for related variables, such as _POSIX_TRACE_NAME_MAX.

Since the stub does not model explicitly which options are supported or not, it seems that the stub may be underapproximating (by not returning nondet results in cases like this).

I agree that the other requirements can be abstracted
> “the value shall not be more restrictive than the compile-time value”
> “the value shall not change during system execution”